src/devices/Ad7193/Ad7193.cs

```diff
@@ -0,0 +1,838 @@
+        #endregion
+
+        #region Metadata for ISpiDevice
+
+        /// <summary>
+        /// The list of SPI modes that are valid for this device
+        /// </summary>
+        public const SpiMode ValidSpiModes = SpiMode.Mode3;
+
+        /// <summary>
+        /// The maximum frequency that can be used on the SPI bus
+        /// </summary>
+        public const int MaximumSpiFrequency = 10000000;
+        #endregion
+
+        #region metadata for IAdcDevice
+
+        /// <summary>
+        /// Number of ADCs on the device
+        /// </summary>
+        public const int ADCCount = 1;
+
+        /// <summary>
+        /// The maximum bitrate of the ADC
+        /// </summary>
+        public const int ADCBitrate = 24;
+
+        /// <summary>
+        /// The maximum sampling rate of the ADC
+        /// </summary>
+        public const int ADCSamplerate = 4800;
+
+        /// <summary>
+        /// The number of channels the ADC has
+        /// </summary>
+        public const int ADCInputChannelCount = 8;
+        #endregion
+
+        private readonly object _spiTransferLock = new object();
+        private readonly Stopwatch _stopWatch = new Stopwatch();
+
+        private SpiDevice _spiDevice = null;
+
+        /// <summary>
+        /// The list of received ADC values
+        /// </summary>
+        // protected ConcurrentQueue<AdcValue> adcValues = new ConcurrentQueue<AdcValue>();
+        protected BlockingCollection<AdcValue> adcValues = new BlockingCollection<AdcValue>();
+
+        /// <summary>
+        /// The event that is fired every time a new value is received from the ADC
+        /// </summary>
+        public event EventHandler<AdcValueReceivedEventArgs> AdcValueReceived;
+
+        private readonly StringBuilder _sb = new StringBuilder();
+
+        private readonly uint[] _registerCache = { 0x00, 0x080060, 0x000117, 0x000000, 0xa2, 0x00, 0x000000, 0x000000 };
+        private readonly byte[] _registerSize = { 1, 3, 3, 3, 1, 1, 3, 3 };
+
+        /// <summary>
+        /// The external reference voltage value. The default is 2.5V on REFIN1+ and REFIN1- (on the Digilent Pmod AD5 board)
+        /// </summary>
+        public double VReference { get; set; } = 2.50f;
+
+        /// <summary>
+        /// Gain level on the ADC
+        /// </summary>
+        public Gain PGAGain
+        {
+            set
+            {
+                _registerCache[(byte)Register.Configuration] &= 0xFF_FFF8;          // keep all bit values except Gain bits
+                _registerCache[(byte)Register.Configuration] |= (uint)value;
+
+                SetRegisterValue(Register.Configuration, _registerCache[(byte)Register.Configuration]);
+            }
+        }
+
+        /// <summary>
+        /// Human-readable representation of the current status register
+        /// </summary>
+        public string Status
```

Comment:
Seems like this getter is actually performing a lot of operations so I guess that we should probably move this to be a GetStatus method instead.

Reply:
Status, Mode and Config are all like this, and I was hesitant about their naming as well.
They basically show the register values in a human readable form, they should be named differently.
What do you think? GetStatus or GetStatusText would be more appropriate? 
Or I could even more these into a RegisterToString(Register register) method, and change the Status, Mode and Config properties to uints with their getters and setters.